Sources/BitcoinKit/Core/Mnemonic/Mnemonic+Generate.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  Mnemonic+Generate.swift
+//
+//  Copyright © 2018 BitcoinKit developers
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+// MARK: - Public
+
+#if BitcoinKitXcode
+import BitcoinKit.Private
+#else
+import BitcoinKitPrivate
+#endif
+
+// MARK: Generate
+public extension Mnemonic {
+    static func generate(strength: Strength = .default, language: Language = .english) throws -> [String] {
+        let entropy = try securelyGenerateBytes(count: strength.byteCount)
+        return try generate(entropy: entropy, language: language)
+    }
+}
+
+internal extension Mnemonic {
+    static func generate(
+        entropy: Data,
+        language: Language = .english
+    ) throws -> [String] {
```

Comment:
Do you think we still need to separate `generate(entropy:language:)` method from `generate(strength:language:)`?

I think codes in `generate(entropy:language)` can be moved to `generate(strength:language)`. Then, we don't need to check `Error.unsupportedByteCountOfEntropy`.

Also, I think an implementation like this is more clear for the future reader of this code.
```
extension Mnemonic.Strength {
    var wordCount: Int {
        switch rawValue {
        case 128: return 12 // 132bits = Entropy(128 bit) + Checksum(4 bit)
        case 160: return 15 // 165bits = Entropy(160 bit) + Checksum(5 bit)
        case 192: return 18 // 198bits = Entropy(192 bit) + Checksum(6 bit)
        case 224: return 21 // 231bits = Entropy(224 bit) + Checksum(7 bit)
        case 256: return 24 // 264bits = Entropy(256 bit) + Checksum(8 bit)
        default: fatalError("Unsupported strength")
        }
    }
    var checksumLengthBits: Int {
        return wordCount * 11 - rawValue
    }
}
```